src/main/java/org/variantsync/diffdetective/diff/text/DiffLineNumberRange.java

```diff
@@ -1,4 +1,18 @@
 package org.variantsync.diffdetective.diff.text;
 
+import java.util.Objects;
+
 public record DiffLineNumberRange(DiffLineNumber from, DiffLineNumber to) {
+    @Override
+    public boolean equals(Object o) {
```

Comment:
All records have generated `equals` and `hashCode` (and `toString`) methods to reduce boilerplate (see [The Java® Language Specification](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.10.3)). There is no need to implement them yourself.